File: app/src/main/java/com/tokyonth/installer/widget/CustomizeDialog.java
package com.tokyonth.installer.widget;

import android.content.Context;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.tokyonth.installer.Contents;
import com.tokyonth.installer.R;
import com.tokyonth.installer.utils.SPUtils;

public class CustomizeDialog extends MaterialAlertDialogBuilder {

    public CustomizeDialog(Context context) {
        super(context);
    }

    private CustomizeDialog(Context context, int overrideThemeResId) {
        super(context, overrideThemeResId);
    }

    public static CustomizeDialog getInstance(Context context) {
        int theme = (boolean) SPUtils.getData(Contents.SP_NIGHT_MODE, false) ?
                R.style.DialogTheme : 0;
        return  new CustomizeDialog(context, theme);

    }

}

File: app/src/main/java/com/tokyonth/installer/widget/CustomizeDialog.kt
package com.tokyonth.installer.widget

import android.content.Context

import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.tokyonth.installer.Constants
import com.tokyonth.installer.R
import com.tokyonth.installer.utils.SPUtils

class CustomizeDialog private constructor(context: Context, overrideThemeResId: Int) :
        MaterialAlertDialogBuilder(context, overrideThemeResId) {
    companion object {

        fun getInstance(context: Context): CustomizeDialog {
            val theme = if (SPUtils.getData(Constants.SP_NIGHT_MODE, false) as Boolean)
                R.style.DialogTheme
            else
                0
            return CustomizeDialog(context, theme)
        }
    }

}
